base-compat-batteries/src/Data/List/NonEmpty/Compat.hs

```diff
@@ -126,4 +129,39 @@ tails1 =
   -- - Therefore, if we take all but the last element of `tails xs` i.e.
   --   `init (tails xs)`, we have a nonempty list of nonempty lists
   fromList . Prelude.map fromList . List.init . List.tails . Foldable.toList
+
+-- | A monomorphic version of '<>' for 'NonEmpty'.
+--
+-- >>> append (1 :| []) (2 :| [3])
+-- 1 :| [2,3]
+--
+-- /Since: 4.16/
+append :: NonEmpty a -> NonEmpty a -> NonEmpty a
+append = (<>)
```

Comment:
I think this needs to be `(Prelude.<>)` due to the use of `NoImplicitPrelude`. (Similarly for other uses of `<>` here and in `base-compat-batteries`.)